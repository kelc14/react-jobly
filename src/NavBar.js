import React from "react";
import { NavLink, Link } from "react-router-dom";
import "./NavBar.css";

const NavBar = () => {
  return (
    <ul className="NavBar">
      <li className="NavBar-li">
        <Link to="/" className="Jobly-logo-txt">
          JOBLY
        </Link>
      </li>

      <li className="NavBar-li" style={{ float: "right" }}>
        <NavLink to="/profile ">Profile</NavLink>
      </li>
      <li className="NavBar-li" style={{ float: "right" }}>
        <NavLink to="/signup ">Sign Up</NavLink>
      </li>
      <li className="NavBar-li" style={{ float: "right" }}>
        <NavLink to="/login ">Log in</NavLink>
      </li>

      <li className="NavBar-li" style={{ float: "right" }}>
        <NavLink to="/companies">Companies</NavLink>
      </li>
      <li className="NavBar-li" style={{ float: "right" }}>
        <NavLink to="/jobs">Jobs</NavLink>
      </li>
    </ul>
  );
};

export default NavBar;
