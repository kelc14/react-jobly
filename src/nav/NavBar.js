import React, { useContext } from "react";
import { NavLink, Link } from "react-router-dom";
import "./NavBar.css";
import UserContext from "../hooks/UserContext";

/**
 * NavBar Component ->
 *
 * if username stored (auth) - display authenticated routes:
 * -jobs, companies, profile, logout
 *
 * if anon - display anon routes
 * - log in, sign up
 *
 */

const NavBar = ({ logoutUser }) => {
  const user = useContext(UserContext);

  const authLinks = () => {
    return (
      <>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink
            to="/ "
            onClick={logoutUser}
          >{`Logout ${user.username}`}</NavLink>
        </li>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink to="/profile ">Profile</NavLink>
        </li>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink to="/companies">Companies</NavLink>
        </li>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink to="/jobs">Jobs</NavLink>
        </li>
      </>
    );
  };

  const anonLinks = () => {
    return (
      <>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink to="/signup ">Sign Up</NavLink>
        </li>
        <li className="NavBar-li" style={{ float: "right" }}>
          <NavLink to="/login ">Log in</NavLink>
        </li>
      </>
    );
  };

  return (
    <ul className="NavBar">
      <li className="NavBar-li">
        <Link to="/" className="Jobly-logo-txt">
          JOBLY
        </Link>
      </li>

      {user ? authLinks() : anonLinks()}
    </ul>
  );
};

export default NavBar;
