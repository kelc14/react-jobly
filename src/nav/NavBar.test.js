import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import NavBar from "./NavBar";

/**  SMOKE TEST */
it("renders without crashing", function () {
  render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
});

/**  SNAPSHOT TEST */
it("renders without crashing", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <NavBar />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
