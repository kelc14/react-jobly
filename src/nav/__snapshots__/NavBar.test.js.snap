// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <ul
    class="NavBar"
  >
    <li
      class="NavBar-li"
    >
      <a
        class="Jobly-logo-txt"
        href="/"
      >
        JOBLY
      </a>
    </li>
    <li
      class="NavBar-li"
      style="float: right;"
    >
      <a
        class=""
        href="/signup "
      >
        Sign Up
      </a>
    </li>
    <li
      class="NavBar-li"
      style="float: right;"
    >
      <a
        class=""
        href="/login "
      >
        Log in
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
